SignalServiceKit/src/Messages/OWSBatchMessageProcessor.m

```diff
@@ -471,7 +489,8 @@ - (void)enqueueEnvelopeData:(NSData *)envelopeData
 
     // We need to persist the decrypted envelope data ASAP to prevent data loss.
     [self.processingQueue enqueueEnvelopeData:envelopeData plaintextData:plaintextData transaction:transaction];
```

Comment:
Ah - good find! 

I think that your fix works, but dbModified happens all the time, even though this isn't a particularly expensive method, I think we should try to avoid falling back on observing dbModified when there's another straight forward way. How about instead using `transaction`'s completion queue method:

```
- (void)addCompletionQueue:(dispatch_queue_t)completionQueue
           completionBlock:(dispatch_block_t)completionBlock
```

Reply:
Nice, done.